provider/core.rb

```diff
@@ -394,6 +399,10 @@ def generate_file(data)
     end
 
     def generate_file_write(ctx, data)
+      if File.exist?(data[:out_file]) && File.mtime(data[:out_file]) > @start_time
```

Comment:
Might be handy to have a comment explaining why we're checking file time here for future readers of this code.